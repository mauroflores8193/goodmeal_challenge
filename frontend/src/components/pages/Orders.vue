<template>
  <div class="flex flex-col h-full">
    <Header title="Mis Ã³rdenes"/>
    <div class="px-6 flex flex-col gap-4 flex-1 overflow-y-auto">
      <router-link v-for="(order, index) in orders" :key="index" :to="'/order-detail/'+order.id">
        <OrderCard v-bind="order"/>
      </router-link>
    </div>
    <Nav/>
  </div>
</template>

<script>
import Header from "../molecules/Header.vue";
import OrderCard from "../organisms/OrderCard.vue";
import Nav from "../molecules/Nav.vue";

export default {
  name: "Orders",
  components: { OrderCard, Header, Nav },
  data: function () {
    return {
      orders: [
        {
          id: 466044,
          storeName: 'Fork Bilbao',
          total: 30500,
          pickUpTime: '09:00 - 16:00 hrs',
          pickUpDate: '25/11/22',
        },
        {
          id: 466044,
          storeName: 'Fork Bilbao',
          total: 30500,
          pickUpTime: '09:00 - 16:00 hrs',
          pickUpDate: '25/11/22',
        }
      ]
    }
  }
}
</script>