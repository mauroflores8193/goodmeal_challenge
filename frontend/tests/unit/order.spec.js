import { createLocalVue, mount } from '@vue/test-utils'
import VueRouter from 'vue-router'
import Orders from '@/components/pages/Orders.vue'
import flushPromises from 'flush-promises'
import axios from 'axios'
import orderList from './../mocks/orderList.json'

jest.mock('axios')

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()
test('Orders.vue', async () => {
  axios.get.mockResolvedValue({ data: orderList });
  const tree = mount(Orders, { localVue, router })
  await flushPromises();
  expect(tree).toMatchSnapshot();
})
