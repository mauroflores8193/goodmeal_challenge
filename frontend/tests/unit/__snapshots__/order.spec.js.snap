// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`increments value on click 1`] = `
<div class="flex flex-col h-full">
  <div class="h-16 flex relative items-center">
    <!---->
    <h1 class="text-lg font-medium mx-auto">Mis órdenes</h1>
  </div>
  <div class="px-6 flex flex-col gap-4 flex-1 overflow-y-auto"><a href="#/order-detail/3" class="">
      <!----></a><a href="#/order-detail/2" class="">
      <!----></a><a href="#/order-detail/1" class="">
      <!----></a></div>
  <nav class="flex h-20 justify-around items-center"><a href="#/" aria-current="page" class="router-link-exact-active router-link-active">
      <div class="button-icon text-pink-500">
        <!---->
        <!---->
        <div>Inicio</div>
      </div>
    </a> <a href="#/orders" class="">
      <div class="button-icon">
        <!---->
        <!---->
        <div>Órdenes</div>
      </div>
    </a> <a href="#/profile" class="">
      <div class="button-icon">
        <!---->
        <!---->
        <div>Perfil</div>
      </div>
    </a></nav>
</div>
`;
